lib/rexml/text.rb

```diff
@@ -268,7 +268,10 @@ def inspect
     #   u = Text.new( "sean russell", false, nil, true )
     #   u.value   #-> "sean russell"
     def value
-      @unnormalized ||= Text::unnormalize( @string, doctype )
+      entity_expansion_text_limit =
+        document&.entity_expansion_text_limit || Security.entity_expansion_text_limit
```

Comment:
Ah, we want to avoid this when `@unnormalized` isn't `nil`.

How about this?

```ruby
    def Text::unnormalize( string, doctype=nil, filter=nil, illegal=nil, entity_expansion_text_limit: nil )
      entity_expansion_text_limit ||= Security.entity_expansion_text_limit
```

and

```ruby
@unnormalized ||= Text::unnormalize(@string, doctype, entity_expansion_text_limit: document&.entity_expansion_text_limit)
```

Reply:
OK
I see.
It is nice!